packages/main/src/plugin/troubleshooting.ts

```diff
@@ -0,0 +1,113 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import AdmZip from 'adm-zip';
+import moment from 'moment';
+import * as os from 'node:os';
+import * as fs from 'node:fs';
+
+const SYSTEM_FILENAME = 'system';
+
+export interface FileMap {
+  fileName: string;
+  content: string;
+}
+
+export type LogType = 'log' | 'warn' | 'trace' | 'debug' | 'error';
+
+export class Troubleshooting {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  constructor(private apiSender: any) {}
+
+  // The "main" function that is exposes that is used to gather
+  // all the logs and save them to a zip file.
+  // this also takes in the console logs and adds them to the zip file (see preload/src/index.ts) regarding memoryLogs
+  async saveLogs(console: { logType: LogType; message: string }[], destination: string): Promise<string[]> {
+    const systemLogs = await this.getSystemLogs();
+    const consoleLogs = this.getConsoleLogs(console);
+    const fileMaps = [...systemLogs, ...consoleLogs];
+    await this.saveLogsToZip(fileMaps, destination);
+    return fileMaps.map(fileMap => fileMap.fileName);
+  }
+
+  async saveLogsToZip(fileMaps: FileMap[], destination: string): Promise<void> {
+    if (fileMaps.length === 0) return;
+
+    const zip = new AdmZip();
+    fileMaps.forEach(fileMap => {
+      zip.addFile(`${fileMap.fileName}.txt`, Buffer.from(fileMap.content, 'utf8'));
```

Comment:
Assumes every filename needs .txt, but maybe we want to include a .log or some other file in the future. Would be better if functions like getFileSystemContent() or generateLogFileName() set the filename they want and it just includes it here.